packages/ketcher-macromolecules/src/hooks/useIDTAliasesTextForMonomer.ts

```diff
@@ -0,0 +1,81 @@
+import {
+  AttachmentPointName,
+  IKetIdtAliases,
+  PolymerBond,
+  Sugar,
+} from 'ketcher-core';
+import { useMemo } from 'react';
+
+type Props = {
+  idtAliases: IKetIdtAliases | undefined;
+  attachmentPointsToBonds?: Record<AttachmentPointName, PolymerBond | null>;
+};
+
+const useIDTAliasesTextForMonomer = ({
+  idtAliases,
+  attachmentPointsToBonds,
+}: Props) => {
+  return useMemo(() => {
+    if (!idtAliases) {
+      return null;
+    }
+
+    const { base, modifications } = idtAliases;
+
+    // For preview on canvas
+    if (attachmentPointsToBonds) {
+      if (!modifications) {
+        return null;
+      }
+
+      const { endpoint5, internal, endpoint3 } = modifications;
+
+      const { R1, R2 } = attachmentPointsToBonds;
+      // Handle phosphate exclusively
+      if (base === 'Phos') {
```

Comment:
Can we use check that monomer instanceof Phosphate here? I believe this rule applicable for all phosphates

Reply:
We don't have a specific monomer instance here, I used MonomerClass instead, I think it should do the trick as well for all the phosphates